### PR TITLE
Issue #11214: Specified violation message in LeftCurlyCheck

### PR DESCRIPTION
**Pull request by kaungmyat-htet**
Issue #11214

### CODE REVIEW

**Comment by Kevin222004** [comment]
You have escaped that character which is not required here because it is not a special regex character.
if you finding difficulties to understand which message to write go to the https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/com/puppycrawl/tools/checkstyle/checks/blocks/messages.properties

**Comment by kaungmyat-htet** [comment]
Thank you very much! That solves the error.

But when the line is longer than 100 character.
I wrap the violation part to the next line.
```java
static Runnable r3 = () -> {String.valueOf("Hello world two!");};
    // violation ''{' at column 32 should be on a new line'
```
 I also change the expected string in the test.
```java
final String[] expected = {
            "18:32: " + getCheckMessage(MSG_KEY_LINE_NEW, "{", 32),
            "25:32: " + getCheckMessage(MSG_KEY_LINE_NEW, "{", 32),
        };
```
But I am still getting these error.

```
[ERROR] testNewLineOptionWithLambda  Time elapsed: 0.141 s  <<< FAILURE!
com.google.common.truth.AssertionErrorWithFacts:
Violation lines for checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\blocks\leftcurly\InputLeftCurlyTestNewLineOptionWithLambda.java differ.
missing (1)   : 25
unexpected (1): 24
---
expected      : [18, 25]
but was       : [18, 24]
        at com.puppycrawl.tools.checkstyle.checks.blocks.LeftCurlyCheckTest.testNewLineOptionWithLambda(LeftCurlyCheckTest.java:315)

```

**Comment by Kevin222004** [comment]
See you can't  wrap the // violation message, to solve this error look up at https://github.com/checkstyle/checkstyle/pull/11342#discussion_r815335679

**Comment by Kevin222004** [comment]
Change your pr description

**Comment by kaungmyat-htet** [comment]
Can I reduce this class name to fit 100 characters?

```java
public class InputLeftCurlyTestSwitchExpressionsNewLine {
    // violation ''{' at column 57 should be on a \n'
```

**Comment by romani** [comment]
No, but you can use comment
`// violation below ...... `

**Comment by kaungmyat-htet** [comment]
@Kevin222004 and @romani 
Thank you very much for helping me out.

**Comment by Kevin222004** [comment]
look at the previous commit and change pr description

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestSwitchExpressionsNewLine.java [hunk 1]
```diff
@@ -14,8 +14,9 @@
 //non-compiled with javac: Compilable with Java14
 package com.puppycrawl.tools.checkstyle.checks.blocks.leftcurly;
 
-public class InputLeftCurlyTestSwitchExpressionsNewLine { // violation
-    int howMany1(int k) { // violation
+// violation below ''{' at column 57 should be on a new line'
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestSwitchExpressionsNewLine.java:17]: Please make comment on line 16, in this case you do not need to shift all other violations in expected array.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyIgnoreEnumsOptTrue.java [hunk 2]
```diff
@@ -18,7 +18,8 @@ enum Colors {RED, // ok
         BLUE,
         GREEN;
 
-        @Override public String toString() { return ""; }; // violation
+        @Override public String toString() { return ""; };
+        // violation above ''{' at column 44 should have line break after'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyIgnoreEnumsOptTrue.java:22]: Please do `//violation below` on empty line above.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestLineBreakAfter.java [hunk 3]
```diff
@@ -55,18 +56,20 @@ else if (x < 0) {
             do {} while (x == 2);
         }
 
-        for (int k = 0; k < 1; k++) { String innerBlockVariable = ""; } // violation
+        // violation below ''{' at column 37 should have line break after'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestLineBreakAfter.java:59]: Move violation comment above instead on empty line

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestLineBreakAfter.java [hunk 4]
```diff
@@ -19,34 +19,35 @@
...
+                { // violation ''{' at column 17 should be on the previous line'
                 case 0:
                     break;
                 default:
                     break;
                 }
             }
-            catch (Exception e) { break; } // violation
-            finally { break; } // violation
+            // violation below ''{' at column 33 should have line break after'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestLineBreakAfter.java:48]: Put comment on line above violation

**Review by romani — CHANGES_REQUESTED** [review]
I commented few but please apply same to all other cases, shifting lines in input file is last resort

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestSwitchExpressionsNewLine.java (hunk 1 above)
Comment (kaungmyat-htet) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestSwitchExpressionsNewLine.java:16]: Done

**Comment by romani** [comment]
Please reply done to each review point

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyIgnoreEnumsOptTrue.java (hunk 2 above)
Comment (kaungmyat-htet) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyIgnoreEnumsOptTrue.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestLineBreakAfter.java (hunk 3 above)
Comment (kaungmyat-htet) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestLineBreakAfter.java:58]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestLineBreakAfter.java (hunk 4 above)
Comment (kaungmyat-htet) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestLineBreakAfter.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestDefault3.java [hunk 5]
```diff
@@ -78,28 +78,29 @@ else if (x < 0) { // ok
...
         boolean flag = true;
         if (flag) { // ok
             System.identityHashCode("heh");
             flag = !flag; } String.CASE_INSENSITIVE_ORDER.
               equals("Xe-xe");
         // it is ok to have rcurly on the same line as previous
         // statement if lcurly on the same line.
-        if (flag) { String.CASE_INSENSITIVE_ORDER.equals("it is ok."); } // violation
+        if (flag) { String.CASE_INSENSITIVE_ORDER.equals("it is ok."); }
+        // violation above ''{' at column 19 should have line break after'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestDefault3.java:103]: Remove 2 comments above and do `violation below `

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheckTest.java [hunk 6]
```diff
@@ -167,16 +167,16 @@ public void testDefault3() throws Exception {
...
-            "114:5: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 5),
-            "123:1: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 1),
-            "125:5: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 5),
-            "134:1: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 1),
-            "136:5: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 5),
-            "138:9: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 9),
-            "153:1: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 1),
-            "162:1: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 1),
-            "169:5: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 5),
+            "112:1: " + getCheckMessage(MSG_KEY_LINE_PREVIOUS, "{", 1),
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheckTest.java:170]: Please do all to avoid shifts in violations

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestDefault3.java (hunk 5 above)
Comment (kaungmyat-htet) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyTestDefault3.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheckTest.java (hunk 6 above)
Comment (kaungmyat-htet) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheckTest.java]: Done

**Review by romani — APPROVED** [review]
Please do one more issue with messages to make sure you understand it and deal with it easily

**Comment by kaungmyat-htet** [comment]
Yes, I am currently working on right curly check which is similar.
But I find difficulty with this long message.
```java
line.same=''{0}'' at column {1} should be on the same line as the next part of \
          a multi-block statement (one that directly contains multiple blocks: \
          if/else-if/else, do/while or try/catch/finally).
```
Can I shorten that message to this according to the situation like below.
```java
try
{

} // violation ''}' at column 13 should be on the same line with catch'
catch (Exception e)
{
    return;
}

if (x > 0)
{
    break;
} // violation ''}' at column 17 should be on the same line with else if'
else if (x < 0) {
      ;
} // violation ''}' at column 17 should be on the same line with else'
else
{
     break;
}
```
I am trying my best to avoid shifts in violation message but
I find a bit difficult in this file https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestWithAnnotations.java

In that file, I am having trouble adding the violation message for second line of code between these long lines of code.
There are no empty lines between so I can't use `// violation below or above`
```java
new Object() { public int hashCode() { return 1; }  { int a = 5; }}; // 2 violations
        new Object() { public int hashCode() { return 1; }  int b = 10; }; // violation
        new Object() { public int hashCode() { return 1; }  { int c = 5; } int d = 8; };
        // 2 violations above

BufferedReader br2 = new BufferedReader(br1)) {} catch (IOException e) { ; }
        try (BufferedReader br1 = new BufferedReader(null); // violation above
                BufferedReader br2 = new BufferedReader(br1)) {
            ;
```

**Comment by romani** [comment]
Yes, we can shorten messages to let them be single line and still understandable

**Pull request merged**